applications/GeoMechanicsApplication/tests/cpp_tests/test_time_incrementor.cpp

```diff
@@ -0,0 +1,439 @@
+    {
+        has_thrown = true;
+    }
+
+    KRATOS_EXPECT_FALSE(has_thrown) // No other way to check that the constructor does not throw
+}
+
+KRATOS_TEST_CASE_IN_SUITE(AdaptiveTimeIncrementorThrowsWhenIncreaseFactorIsSmallerThanOne, KratosGeoMechanicsFastSuite)
+{
+    const auto start_time        = 0.0;
+    const auto end_time          = start_time + 8.0;
+    const auto start_increment   = 0.5;
+    const auto max_num_of_cycles = std::size_t{8};
+    const auto reduction_factor  = 0.5;
+    const auto increase_factor   = 0.5;
+
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(AdaptiveTimeIncrementor(start_time, end_time, start_increment, max_num_of_cycles, reduction_factor, increase_factor),
+                                      "Increase factor must be greater than or equal to 1, but got 0.5")
+}
+
+KRATOS_TEST_CASE_IN_SUITE(AdaptiveTimeIncrementorDoesNotThrowWhenIncreaseFactorEqualsOne, KratosGeoMechanicsFastSuite)
+{
+    const auto start_time        = 0.0;
+    const auto end_time          = start_time + 8.0;
+    const auto start_increment   = 0.5;
+    const auto max_num_of_cycles = std::size_t{8};
+    const auto reduction_factor  = 0.5;
+    const auto increase_factor   = 1.0;
+    auto       has_thrown        = false;
+
+    try
+    {
+        AdaptiveTimeIncrementor(start_time, end_time, start_increment, max_num_of_cycles, reduction_factor, increase_factor);
+    }
+    catch (...)
+    {
+        has_thrown = true;
+    }
+
+    KRATOS_EXPECT_FALSE(has_thrown) // No other way to check that the constructor does not throw
+}
+
+KRATOS_TEST_CASE_IN_SUITE(AdaptiveTimeIncrementorThrowsWhenMaxNumberOfIterationsEqualsZero, KratosGeoMechanicsFastSuite)
+{
+    const auto start_time            = 0.0;
+    const auto end_time              = start_time + 8.0;
+    const auto start_increment       = 0.5;
+    const auto max_num_of_cycles     = std::size_t{8};
+    const auto reduction_factor      = 0.5;
+    const auto increase_factor       = 2.0;
+    const auto min_num_of_iterations = std::size_t{0};
+    const auto max_num_of_iterations = std::size_t{0};
+
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(AdaptiveTimeIncrementor(start_time, end_time, start_increment, max_num_of_cycles, reduction_factor, increase_factor, min_num_of_iterations, max_num_of_iterations),
+                                      "Minimum number of iterations (0) is not less than maximum number of iterations (0)")
+}
+
+KRATOS_TEST_CASE_IN_SUITE(AdaptiveTimeIncrementorThrowsWhenMinNumberOfIterationsIsGreaterThanMaxNumberOfIterations, KratosGeoMechanicsFastSuite)
+{
+    const auto start_time            = 0.0;
+    const auto end_time              = start_time + 8.0;
+    const auto start_increment       = 0.5;
+    const auto max_num_of_cycles     = std::size_t{8};
+    const auto reduction_factor      = 0.5;
+    const auto increase_factor       = 2.0;
+    const auto max_num_of_iterations = std::size_t{10};
+    const auto min_num_of_iterations = max_num_of_iterations + 1;
```

Comment:
I think it would be nice to have valid defaults for the inputs, such that you don't have to specify all the valid inputs every test (and it's clear in the test which variable is specifically tested.

Reply:
Agreed. I have added a `struct` to hold the time incrementor's settings. A default-constructed instance of that `struct` constitutes a valid collection of settings. Furthermore, I have added a convenience function to build an adaptive time incrementor given a settings object. Most tests are now more compact and to the point. Thanks for the suggestion!